### PR TITLE
Add publish action

### LINKED ISSUE #3: publish on npm
We should setup a proper release to the npm registry to support version constrains in consuming projects.

### PR DESCRIPTION
**Pull request by jlvandenhout**
I've set up an `iota-community` organization on npmjs. This should publish to both the npmjs registry and this repos package registry when a release tag is pushed.

Closes #3.

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    - run: npm install
+    - run: npm publish --access public
+      env:
+        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+    - uses: actions/setup-node@v2
+      with:
+        registry-url: 'https://npm.pkg.github.com'
+    - run: npm publish --access public
+      env:
+        NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Dr-Electron) [inline .github/workflows/publish.yaml:23]:
```suggestion
        NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}

```

Comment (Dr-Electron) [reply .github/workflows/publish.yaml]: Please add a new line :trollface:

**Review by Dr-Electron — CHANGES_REQUESTED** [review]

**Review by Dr-Electron — APPROVED** [review]
Lgtm 🤔

**Review by eike-hass — CHANGES_REQUESTED** [review]
we should ignore the lib folder in the `.gitignore` since it will be build and pushed to the registries.

**Comment by eike-hass** [comment]
I wonder what the correct workflow for a release is. Maybe:
0. make sure signed commits are enabled locally
1. run `npm version X.X.X`
2. push changes (including the new tag)
3. create GitHub release
4. profit

**Comment by eike-hass** [comment]
@jlvandenhout could you resign your commits? Let me know if you need help with that 👍

**Review by lucas-tortora — APPROVED** [review]

**Comment by eike-hass** [comment]
> @jlvandenhout could you resign your commits? Let me know if you need help with that 👍

resolved :)

**Comment by eike-hass** [comment]
> I wonder what the correct workflow for a release is. Maybe:
> 
> 1. make sure signed commits are enabled locally
> 2. run `npm version X.X.X`
> 3. push changes (including the new tag)
> 4. create GitHub release
> 5. profit
> 
> If yes, let's put it in the Readme

Will document after the first successful release.

**Pull request merged**